sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/ChangeFeedProcessorBuilderImpl.java

```diff
@@ -313,30 +316,26 @@ public ChangeFeedProcessorBuilderImpl(PartitionManager partitionManager) {
     }
 
     private Mono<Void> initializeCollectionPropertiesForBuild() {
-        ChangeFeedProcessorBuilderImpl self = this;
-
         if (this.changeFeedProcessorOptions == null) {
             this.changeFeedProcessorOptions = new ChangeFeedProcessorOptions();
         }
 
         return this.feedContextClient
             .readDatabase(this.feedContextClient.getDatabaseClient(), null)
             .map( databaseResourceResponse -> {
-                self.databaseResourceId = databaseResourceResponse.database().id();
-                return self.databaseResourceId;
+                this.databaseResourceId = databaseResourceResponse.database().id();
```

Comment:
same here, this will be executed when a consumer subscribes to the returned Mono from this method.
which means this could happen on a different thread.

now consider a hypothetical scenario in which a second thread wants to access this.databaseResourceid. as none of the class fields are neither atomic nor volatile I wonder if the second thread can see the restult set by the first thread?

Reply:
The databaseResourceId corresponds to the database name set by the user when ChangeFeedProcessor is built and there's at most only one id that can be set for the current object. Also ChangeFeedProcessor is an immutable object so the database name will not change after being set.

Reply:
databaseId cannot change but for the first time can be set.

Please see my general comment on thread-safely in accessing non synchronized/atomic class fields on multiple threads.

Reply:
"volatile" has been added to the offending class field.